packages/snfoundry/scripts-ts/deploy-contract.ts

```diff
@@ -49,46 +51,73 @@ const deployContract_NotWait = async (payload: {
   classHash: string;
   constructorCalldata: RawArgs;
 }) => {
-  let { calls, addresses } = transaction.buildUDCCall(
-    payload,
-    deployer.address
-  );
-  deployCalls.push(...calls);
-  return {
-    contractAddress: addresses[0],
-  };
+  try {
+    const { calls, addresses } = transaction.buildUDCCall(
+      payload,
+      deployer.address
+    );
+    deployCalls.push(...calls);
+    return {
+      contractAddress: addresses[0],
+    };
+  } catch (error) {
+    console.error(red("Error building UDC call:"), error);
+    throw error;
+  }
 };
 
+/**
+ * Deploy a contract using the specified parameters.
+ *
+ * @param {DeployContractParams} params - The parameters for deploying the contract.
+ * @param {string} params.contract - The name of the contract to deploy.
+ * @param {string} [params.contractName] - The name to export the contract as (optional).
+ * @param {RawArgs} [params.constructorArgs] - The constructor arguments for the contract (optional).
+ * @param {UniversalDetails} [params.options] - Additional deployment options (optional).
+ *
+ * @returns {Promise<{ classHash: string; address: string }>} The deployed contract's class hash and address.
+ *
+ * @example
+ * ///Example usage of deployContract function
+ * await deployContract({
+ *   contract: "YourContract",
+ *   contractName: "YourContractExportName",
+ *   constructorArgs: { owner: deployer.address },
+ *   options: { maxFee: BigInt(1000000000000) }
+ * });
+ */
 const deployContract = async (
-  constructorArgs: RawArgs,
-  contractName: string,
-  exportContractName?: string,
-  options?: {
-    maxFee: bigint;
-  }
+  params: DeployContractParams
 ): Promise<{
   classHash: string;
   address: string;
 }> => {
+  const { contract, constructorArgs, contractName, options } = params;
+  deploymentOptions = options;
```

Comment:
this `options.maxFee` is meant to be passed to `declareIfNot_NotWait` as declareOptions. 
this current implementation keeps track of one single `deploymentOptions` which is updated on each call, if this options are not present on a call it will use the previous one.

Reply:
if no options was passed on a new call, options is undefined and overwrites the previously set options.

The reason i am doin it this way is to allow the user set the options for `declareIfNot_NotWait` and `executeDeployCalls`  once, largely because of of how we call the script.

```ts
deployScript()
  .then(async () => {
    await executeDeployCalls();
    exportDeployments();

    console.log(green("All Setup Done"));
  })
  .catch(console.error);
```

but since you have specified that they are different values, i would go ahead and allow users set them independently